src/smtencoding/FStar.SMTEncoding.Solver.fs

```diff
@@ -220,20 +220,12 @@ let ask_and_report_errors env all_labels prefix query suffix =
             | ([], _), _
             | _, Inl _ -> result
             | _, Inr _ -> Inr errs in
+            
         let rec try_alt_configs prev_f (p:decl) (errs:z3_err) cfgs (scope:scope_t) =
             set_minimum_workable_fuel prev_f errs;
             match cfgs, snd errs with
             | [], _
             | _, Kill -> report p errs
-            | [mi], _-> //we're down to our last config; last ditch effort to get a counterexample with very low fuel
-                begin match errs with
-                | [], _ ->
-                    Z3.ask None all_labels (with_fuel [] p mi) (Some scope) (cb false mi p [] scope)
```

Comment:
Removing the case `[mi], _` means that we'll now enter the `mi::tl, _` case below. 
- When `errs = [],_` the behaviour is the same because `use_errors errs result = result`.
- Otherwise, the code used to report `errs` and terminate, while now it asks the last query and reports `errs` if it fails.

The old code seems to assume that the last combination to try is `min_fuel, 1` as a last ditch resort to get a counterexample and doesn't bother asking the query when a counterexample had been found. But this is not the case if `min_fuel = initial_fuel` since the last combination is `max_fuel, max_ifuel`.

An alternative fix will be to unconditionally add `min_fuel, 1` as the last configuration, not only when `min_fuel < initial_fuel`. 

The problem with the original code, this fix and the alternative fix is that a last query with `min_fuel, 1` may succeed, which will be surprising if `max_ifuel < 1`. Making the whole thing fail when this last query succeeds to avoid surprises seems wasteful. 

I have the feeling that `min_fuel` is very hacky and that we should remove it and just live with the fact that we may get no counterexample with a given fuel setting.